examples/classic_controllers/classic_controllers.py

```diff
@@ -648,97 +865,613 @@ def __init__(self, environment, stages, ref_states, color='g', **controller_kwar
+
+
+            psi_i_d_q = np.array(psi_i_d_q)
+            psi_i_d_q_neg = np.rot90(np.array([psi_i_d_q[:, 0], -psi_i_d_q[:, 1], psi_i_d_q[:, 2], -psi_i_d_q[:, 3]]))
+            psi_i_d_q = np.append(psi_i_d_q_neg, psi_i_d_q, axis=0)
+            return np.array(psi_i_d_q)
+
+        self.mtpc = mtpc()
+        self.mtpf = mtpf()
+
+        self.psi_t = np.sqrt(
+            np.power(self.psi_p + self.l_d * self.mtpc[:, 1], 2) + np.power(self.l_q * self.mtpc[:, 2], 2))
+        self.psi_t = np.array([self.mtpc[:, 0], self.psi_t])
+
+        self.i_q_max = np.linspace(-self.nominal_values[self.i_sq_idx], self.nominal_values[self.i_sq_idx], self.i_count)
+        self.i_d_max = -np.sqrt(self.nominal_values[self.i_sq_idx] ** 2 - np.power(self.i_q_max, 2))
+        i_count_mgrid = 200j
+        i_d, i_q = np.mgrid[-self.limit[self.i_sd_idx]:0:i_count_mgrid,
+                            -self.limit[self.i_sq_idx]:self.limit[self.i_sq_idx]:i_count_mgrid / 2]
+        i_d = i_d.flatten()
+        i_q = i_q.flatten()
+        if self.l_d != self.l_q:
+            idx = np.where(np.sign(self.psi_p + i_d * self.l_d) * np.power(self.psi_p + i_d * self.l_d, 2) + np.power(
+                i_q * self.l_q, 2) > 0)
+        else:
+            idx = np.where(self.psi_p + i_d * self.l_d > 0)
+        i_d = i_d[idx]
+        i_q = i_q[idx]
+
+        t = self.p * 1.5 * (self.psi_p + (self.l_d - self.l_q) * i_d) * i_q
+        psi = np.sqrt(np.power(self.l_d * i_d + self.psi_p, 2) + np.power(self.l_q * i_q, 2))
+
+        self.t_min = np.amin(t)
+        self.t_max = np.amax(t)
+
+        self.psi_min = np.amin(psi)
+        self.psi_max = np.amax(psi)
+
+        if torque_control == 'analytical':
+            res = []
+            for psi in np.linspace(self.psi_min, self.psi_max, self.psi_count):
+                ret = []
+                for T in np.linspace(self.t_min, self.t_max, self.t_count):
+                    i_d_, i_q_ = self.solve_analytical(T, psi)
+                    ret.append([T, psi, i_d_, i_q_])
+                res.append(ret)
+            res = np.array(res)
+            self.t_grid = res[:, :, 0]
+            self.psi_grid = res[:, :, 1]
+            self.i_d_inter = res[:, :, 2].T
+            self.i_q_inter = res[:, :, 3].T
+            self.i_d_inter_plot = self.i_d_inter.T
+            self.i_q_inter_plot = self.i_q_inter.T
+
+        elif torque_control == 'interpolate':
+            self.t_grid, self.psi_grid = np.mgrid[np.amin(t):np.amax(t):np.complex(0, self.t_count),
+                                                  self.psi_min:self.psi_max:np.complex(self.psi_count)]
+            self.i_q_inter = griddata((t, psi), i_q, (self.t_grid, self.psi_grid), method='linear')
+            self.i_d_inter = griddata((t, psi), i_d, (self.t_grid, self.psi_grid), method='linear')
+            self.i_d_inter_plot = self.i_d_inter
+            self.i_q_inter_plot = self.i_q_inter
+
```

Comment:
since `torque_control` is user input, maybe add an eventual else clause here raising NotImplementedError for strings different from "analytical" and "interpolate"